src/v/cluster/tm_stm.cc

```diff
@@ -252,6 +252,31 @@ ss::future<checked<model::term_id, tm_stm::op_status>> tm_stm::do_barrier() {
         });
 }
 
+model::record_batch_reader make_checkpoint() {
+    storage::record_batch_builder builder(
+      model::record_batch_type::checkpoint, model::offset(0));
+    builder.add_raw_kv(iobuf(), iobuf());
+    return model::make_memory_record_batch_reader(std::move(builder).build());
+}
+
+ss::future<result<raft::replicate_result>>
+tm_stm::quorum_write_empty_batch(model::timeout_clock::time_point timeout) {
```

Comment:
this method isn't used, why was it added?

Reply:
it is used on line `233`. Since `persisted_stm` is no longer inheriting from `raft::state_machine`